python/ray/experimental/state.py

```diff
@@ -345,21 +346,101 @@ def task_profiles(self):
+            task_info[task_id]["get_task_start"] = event[0]
+          if event[1] == "ray:get_task" and event[2] == 2:
+            task_info[task_id]["get_task_end"] = event[0]
+          if event[1] == "ray:import_remote_function" and event[2] == 1:
+            task_info[task_id]["import_remote_start"] = event[0]
+          if event[1] == "ray:import_remote_function" and event[2] == 2:
+            task_info[task_id]["import_remote_end"] = event[0]
+          if event[1] == "ray:acquire_lock" and event[2] == 1:
+            task_info[task_id]["acquire_lock_start"] = event[0]
+          if event[1] == "ray:acquire_lock" and event[2] == 2:
+            task_info[task_id]["acquire_lock_end"] = event[0]
+          if event[1] == "ray:task:get_arguments" and event[2] == 1:
+            task_info[task_id]["get_arguments_start"] = event[0]
+          if event[1] == "ray:task:get_arguments" and event[2] == 2:
+            task_info[task_id]["get_arguments_end"] = event[0]
+          if event[1] == "ray:task:execute" and event[2] == 1:
+            task_info[task_id]["execute_start"] = event[0]
+          if event[1] == "ray:task:execute" and event[2] == 2:
+            task_info[task_id]["execute_end"] = event[0]
+          if event[1] == "ray:task:store_outputs" and event[2] == 1:
+            task_info[task_id]["store_outputs_start"] = event[0]
+          if event[1] == "ray:task:store_outputs" and event[2] == 2:
+            task_info[task_id]["store_outputs_end"] = event[0]
+          if "worker_id" in event[3]:
+            task_info[task_id]["worker_id"] = event[3]["worker_id"]
+          if "function_name" in event[3]:
+            task_info[task_id]["function_name"] = event[3]["function_name"]
+    return task_info
+
+  def actor_profiles(self):
+    actor_info = dict()
+    actors = self.redis_client.keys("Actor*")
+    for actor in actors:
+      actor_key_str = actor[len('Actor:'):]
+      actor_info['Actor:{}'.format(binary_to_hex(actor))] = self.redis_client.hgetall(actor)
+      x = actor_info['Actor:{}'.format(binary_to_hex(actor))]
+      if b'class_id' in x:
+        x['class_id'] = binary_to_hex(x[b'class_id'])
+        del x[b'class_id']
+      if b'class'in x:
+        x['class'] = binary_to_hex(x[b'class'])
+        del x[b'class']
+      if b'driver_id' in x:
+        x['driver_id'] = binary_to_hex(x[b'driver_id'])
+        del x[b'driver_id']
+      if b'class_name' in x:
+        x['class_name'] = binary_to_hex(x[b'class_name'])
+        del x[b'class_name']
+      if b'module' in x:
+        x[b'module'] = binary_to_hex(x[b'module'])
+        del x[b'module']
+      if b'actor_method_names' in x:
+        x['actor_method_names'] = binary_to_hex(x[b'actor_method_names'])
+        del x[b'actor_method_names']
+      if b'num_gpus' in x:
+        x['num_gpus'] = binary_to_hex(x[b'num_gpus'])
+        del x[b'num_gpus']
+    return actor_info
+
+  def worker_profiles(self):
```

Comment:
This function may be incorrect. I tried it out and one of the entries looked like this.

```
'Workers:1942dcca8cb7cf0c1eb09e51285e12c92464f02d': {'local_scheduler_socket': '2f746d702f7363686564756c65723737383137393630',
  'node_ip_address': '3132372e302e302e31',
  'plasma_manager_socket': '2f746d702f706c61736d615f6d616e616765723134313736323234',
  'plasma_store_socket': '2f746d702f706c61736d615f73746f72653235323038373934',
  'stdout_file': '2f746d702f7261796c6f67732f776f726b65722d3038313237382e6f7574'}
```

These values seem incorrect (e.g., the node IP address).